python/cudf/cudf/core/column/string.py

```diff
@@ -4669,14 +4669,13 @@ def subword_tokenize(
         array([[0, 0, 2],
                [1, 0, 1]], dtype=uint32)
         """
-        warning_message = (
-            "`ser.str.subword_tokenize` API will be depreciated"
-            " in future versions of cudf.\n"
-            "Use `cudf.core.subword_tokenizer.SubwordTokenizer` "
-            "instead"
+        warnings.warn(
+            "`Series.str.subword_tokenize` will be deprecated "
+            "in future versions of cudf.\n"
+            "Use `cudf.core.subword_tokenizer.SubwordTokenizer` instead.",
+            FutureWarning,
```

Comment:
@VibhuJawa Is this feature supposed to _be_ deprecated? That appeared to be the intent of #8603 but this warning indicates _future_ deprecation. It seems like it might be more appropriate to skip `FutureWarning` and just issue a `DeprecationWarning` as the PR title indicated in #8603.

Reply:
> Is  this feature supposed to be deprecated? 

yup,  This feature will be `deprecated ` in next release (gave users a 2 release window to fix) , Have a issue raised [here](https://github.com/rapidsai/cudf/issues/8604).  


> I seems like it might be more appropriate to skip FutureWarning and just issue a DeprecationWarning as the PR title indicated in #8603.  DeprecationWarning is hidden by default but FutureWarning is not. See #9347 (comment)

Yeah,   we did not go with  DeprecationWarning because it is hidden (especially) in `Interactive Notebook` environments.  We discussed this a bit more on the PR [here](https://github.com/rapidsai/cudf/pull/8603#discussion_r658298855). 


TLDR is we followed  pandas and sckit learn as they both use FutureWarning (See [section](https://github.com/rapidsai/cudf/pull/8603) on contributing.md of sckit-learn) . 

Also Related PEP: https://www.python.org/dev/peps/pep-0565/

Reply:
Super helpful info! It sounds like I should switch this PR to use `FutureWarning` instead of `DeprecationWarning`.

As for release cycles, do I understand your intention correctly here?
1. (current release) Announce deprecation.
2. (next release) Deprecate feature.
3. (next-next release) Remove feature.

If that's correct, why not just deprecate in the current release with the comment "this will be removed in next-next"? It sounds like the extra step of announcing a deprecation prior to the deprecation is not needed, since deprecation is an announcement before removal.

Reply:
>  But I don't know what you mean by `deprecated` -- this warning _is_ a deprecation, right? How do you differentiate the announcement of "this will be deprecated" vs. "this is deprecated"?


Thats a good question. Honestly , I dont know the best practice around this. I was  just following [sckit-learn](https://github.com/scikit-learn/scikit-learn/blob/5a879f4a024d499c74da65f0343b535be4aa096e/doc/developers/contributing.rst#deprecation) (among other libraries) of just holding off for two releases  before removing the feature.   

FWIW, `Sckit-learn` does different things depending on what is being deprecated ( i.e different policy for attributes, parameters, changing defaults ). It might be beneficial to standardize this and add to our `Contributing.MD`.  



CC: @vyasr  , @shwina  for guidance on how to go about it.

Reply:
I think a deprecating over one release via `FutureWarning` is fine; two releases out is harder to maintain/track.

1. (upcoming release) Deprecate feature (add a `FutureWarning`) and open a GitHub issue to track its removal.
2. (next release) Remove feature

Reply:
I think there's also just a terminology confusion here. We typically use "deprecation" to the process of marking that something will be removed, and "removal" as the actual release when it is removed. So when @VibhuJawa said "this feature will be deprecated in the next release" he really means that's when it will be removed.

Reply:
An aside that may be of interest and help contextualize the choice of warning: FutureWarning was originally (i.e. in Python 2.7) intended to indicate semantic changes to a user-facing API, not API changes. That encompassed behavioral changes that might otherwise not be noticed by users. It was only after Python 2.7/Python 3.2 made DeprecationWarning silent that a lot of confusion began to arise about the best way to loudly signal deprecations for user-facing features (as opposed to the deprecations that only affect developers, which were the ones that silencing DeprecationWarning was intended to hide). FutureWarning was repurposed for this as a result.

Reply:
Gotcha. I updated the text of this warning. Thanks for the clarification. 😄